build.bash

```diff
@@ -13,25 +13,28 @@ EOF
 	false
 fi
 
+if ! docker version; then
+	echo 'Docker not found' >&2
+	false
+fi
+
+if ! docker buildx version; then
+	echo '"docker buildx" not found (see https://docs.docker.com/buildx/working-with-buildx/ )' >&2
+	false
+fi
+
 IW2SRC="$(realpath "$IW2SRC")"
-BLDCTX="$(realpath "$(dirname "$0")")"
+OUR_DIR="$(realpath "$(dirname "$0")")"
+OUR_FILES="$(mktemp -d)"
 
-docker build -f "${BLDCTX}/action.Dockerfile" -t icinga/icingaweb2-builder "$BLDCTX"
+trap "rm -rf $OUR_FILES" EXIT
 
-docker run --rm -i \
-	-v "${IW2SRC}:/iw2src:ro" \
-	-v "${BLDCTX}:/bldctx:ro" \
-	-v /var/run/docker.sock:/var/run/docker.sock \
-	-e BUILD_MODE \
-	icinga/icingaweb2-builder bash <<EOF
-set -exo pipefail
+pushd "$OUR_FILES"
 
-git -C /iw2src archive --prefix=iw2cp/icingaweb2/ HEAD |tar -xC /
-cd /iw2cp
+git -C "$IW2SRC" archive --prefix=icingaweb2/ HEAD |tar -x
+"${OUR_DIR}/get-mods.sh" "$BUILD_MODE"
+"${OUR_DIR}/composer.bash"
```

Comment:
This is now run outside of any container, i.e. relies on composer version from the host system (if it exists at all, it doesn't on my machine).

Reply:
Yes.

Reply:
Why? I'd try to do as much as possible inside a container to have a more controlled environment (isn't that the very idea why we use containers in the first place?). In [docker-icinga2](https://github.com/Icinga/docker-icinga2/blob/5a7f6b0b9f2ce5e8ba11cf88094b75ddfc56edbf/Dockerfile#L14-L22), even the `git clone` happens inside a container.